sdk/cpp/core/src/service_provider.cpp

```diff
@@ -25,17 +25,12 @@
 //
 //////////////////////////////////////////////////////////////////
 
-#ifndef SERVICE_HPP
-#define SERVICE_HPP
-
-#include <string>
-#include <memory>
+#include "path_api.hpp"
 
 namespace ydk {
-class Service
-{
-};
 
+path::ServiceProvider::~ServiceProvider()
```

Comment:
`ServiceProvider` should be moved out of `path` namespace. This is an important part of this change as we want to decouple `ServiceProvider` completely from path API

`ydk::path::ServiceProvider` --> `ydk::ServiceProvider`